### PR TITLE
ba-80 Replace search header in inventory with search element

### PR DESCRIPTION
**Pull request by lyio**
- Include socobo-element-search
- Change demo page to use new component database according to #ba-75
- Implement some rudimentary WCT tests
- Use socobo-element-search in inventory-overview
- Implement item filtering according to the search
- Implement check all and uncheck all

### CODE REVIEW

**Comment by flashback2k14** [comment]
- layout still broken
- inventory items doesn't load
- unhandled exception
  bug

**Comment by lyio** [comment]
- As you can read, this PR has nothing to do with layout.
- The exception is caused because there are no items available
- Investigating why no items are shown, as the component by itself loads the items fine.

**Comment by lyio** [comment]
Looks like this for me...

socobo-inventory

Once I get bower to install everything the search header should be there as well.

**Comment by lyio** [comment]
socobo-inventory-filtered

Working for me. 

Current project master with this branch as socobo-inventory.

File: socobo-inventory-overview.html [hunk 1]
```diff
@@ -42,20 +42,23 @@
...
       <paper-material vertical layout class="inventory" elevation="1">
-        <paper-toolbar class="searchBox">
-          <paper-checkbox class="checkBoxAll" id="inventoryCheckBoxAll"></paper-checkbox>
-          <paper-input class="search-input" id="inventorySearch" label="Suche"></paper-input>
-        </paper-toolbar>
+        <socobo-element-search 
+          id="searchHeader" 
+          on-check-all-items="_handleCheckAll"
+          on-uncheck-all-items="_handleUncheckAll"
+          on-changed-search-query="_searchQueryChanged">
```
Comment (flashback2k14) [inline socobo-inventory-overview.html]: you have not implemented this event 

``` html
/** 
 * This Event is fired if the search query is cleared. <br/>
 * @event clear-search-query
 */
```

this fires after press the x button.

**Comment by flashback2k14** [comment]
after using master branch for the project. i see inventory items, but i still get this error: 

bug2

**Comment by flashback2k14** [comment]
itemList is a NodeList and a NodeList is not a real array. A simple for loop should fix that bug.

File: socobo-inventory-overview.html (hunk 1 above)
Comment (lyio) [inline socobo-inventory-overview.html]: You said, there was no such event

Comment (flashback2k14) [inline socobo-inventory-overview.html]: me? can't remember. but if you look into my documentation. you see all possible things!

**Pull request merged**